klpbuild/klplib/codestream.py

```diff
@@ -34,7 +34,8 @@ def __init__(self, name, project="", patchid="", kernel="",
         self.project = project
         self.patchid = patchid
         self.kernel = kernel
-        self.archs = archs if archs is not None else []
+
+        self.archs = archs if archs is not None else self.__get_default_archs()
```

Comment:
Good idea! In the past we set the archs later on in the process, even before the Codestream class, but this is much better!